UnicornRender/video/source/Mesh.cpp

```diff
@@ -23,27 +28,33 @@ void Mesh::SetMeshData(const std::vector<Vertex>& vertices, const std::vector<ui
     VerticesUpdated.emit();
 }
 
-void Mesh::SetMaterial(Material material)
+void Mesh::SetMaterial(std::shared_ptr<Material> material)
 {
     m_material = material;
+    m_material->DataUpdated.connect(this, &Mesh::OnMaterialUpdated);
```

Comment:
I think you should also disconnect from old material

Reply:
Thats true, but I when I call DataUpdated.clear(), exception with "violating memory read" is coming. How to bypass this?